src/Symfony/Component/Workflow/Event/EventNameTrait.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Event;
+
+use Symfony\Component\Workflow\Exception\InvalidArgumentException;
+
+/**
+ * @author Nicolas Rigaud <squrious@protonmail.com>
+ */
+trait EventNameTrait
+{
+    /**
+     * Get the event name as lowercase string.
+     *
+     * Example: EnterEvent => enter
+     */
+    public static function getName(): string
```

Comment:
what is the goal of this method ?

Reply:
It is to compute the "simple" name of the event. I thought there could be use cases where ones could extend the workflow and implement their own events.  But if all of this is not intended to be extended, this code could be merged in the computeName method above